app/jobs/worker_status.rb

```diff
@@ -0,0 +1,89 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+#
+# This utility class is used by delayed job workoff rake tasks and allows those
+# to exit when a job completes and the worker is not on the latest task
+# definition (i.e. the latest code)
+class WorkerStatus
+  def conditional_exit!
+    return unless i_am_a_workoff_worker
+
+    Rails.logger.info "This tasks's task definition version: #{my_version}"
+    Rails.logger.info "The latest task definition version: #{latest_version}"
+
+    return if my_version == -1
+    return if latest_version == -1
+
+    return unless my_version < latest_version
+
+    return unless latest_deployment_is_at_least_partially_finished
+
+    Rails.logger.info 'The most recently attempted deployment is finished'
+
+    Rails.logger.warn "Exiting because I am a workoff worker that isn't on the latest version"
+    exit!(0)
+  end
+
+  private
+
+  def latest_deployment_is_at_least_partially_finished
+    return true if Rails.env.development? || Rails.env.test?
+
+    # Couldn't get Curl.get to work
```

Comment:
did you `require 'curb'`?

Reply:
I didn't, but the error wasn't that the class didn't exist. That's an odd way for it to fail if that's all I needed. Want me to try it out?

Reply:
Let's leave it for now.